python/tank/platform/bundle.py

```diff
@@ -529,13 +529,28 @@ def __resolve_hook_path(self, settings_name, hook_expression):
             raise TankError("%s config setting %s: Configuration value cannot be None!" % (self, settings_name))
         
         path = None
+
+        # make sure to replace the engine token if it exists.
+        if constants.TANK_HOOK_ENGINE_REFERENCE_TOKEN in hook_expression:
+            engine_name = self._get_engine_name()
+            if not engine_name:
+                raise TankError(
+                    "No engine could be determined for hook expression '%s'. "
+                    "The hook could not be resolved." % (hook_expression,))
+            else:
+                hook_expression = hook_expression.replace(
+                    constants.TANK_HOOK_ENGINE_REFERENCE_TOKEN,
+                    engine_name,
+                )
         
         # first the default case
```

Comment:
This is the legacy case now, "old-style" hooks. I think that might be worth adding here for clarity